src/lib/assets/hooks/tokens.ts

```diff
@@ -3,76 +3,185 @@ import { useMemo } from 'react';
 
+import { EMPTY_FROZEN_OBJ } from '../../utils';
 import { PREDEFINED_TOKENS_METADATA } from '../known-tokens';
 import type { AccountAsset } from '../types';
-import { tokenToSlug } from '../utils';
+import { toChainAssetSlug, tokenToSlug } from '../utils';
 
-import { isAssetStatusIdle, getAssetStatus } from './utils';
+import { getAssetStatus, isAssetStatusIdle } from './utils';
 
 interface AccountToken extends AccountAsset {
+  chainId: string | number;
   predefined?: boolean;
 }
 
-export const useAllAvailableTokens = (account: string, chainId: string) => {
-  const tokens = useAccountTokens(account, chainId);
-  const allTokensStored = useAllTokensSelector();
-
-  return useMemo(() => {
-    const remainedTokens: AccountToken[] = [];
-    const removedSlugs: string[] = [];
-
-    for (const token of tokens) {
-      if (token.status === 'removed') removedSlugs.push(token.slug);
-      else remainedTokens.push(token);
-    }
-
-    const otherAccountsTokens: AccountToken[] = [];
-    for (const [key, record] of Object.entries(allTokensStored)) {
-      if (isAccountAssetsStoreKeyOfSameChainIdAndDifferentAccount(key, account, chainId))
-        for (const [slug, asset] of Object.entries(record)) {
-          if (asset.status !== 'removed' && !removedSlugs.includes(slug))
-            otherAccountsTokens.push({ slug, status: 'disabled' });
-        }
-    }
-
-    // Keep this order to preserve correct statuses & flags
-    const concatenated = remainedTokens.concat(otherAccountsTokens);
-
-    return sortBy(
-      uniqBy(concatenated, t => t.slug),
-      TOKENS_SORT_ITERATEES
-    );
-  }, [tokens, allTokensStored, account, chainId]);
+export const useEnabledAccountChainTokenSlugs = (accountTezAddress: string, accountEvmAddress: HexString) => {
+  const tezTokens = useEnabledTezosAccountTokenSlugs(accountTezAddress);
+  const evmTokens = useEnabledEvmAccountTokenSlugs(accountEvmAddress);
+
+  return useMemo(() => [...tezTokens, ...evmTokens], [tezTokens, evmTokens]);
+};
+
+export const useAllAccountChainTokensSlugs = (accountTezAddress: string, accountEvmAddress: HexString) => {
+  const tezTokens = useAllTezosAccountTokenSlugs(accountTezAddress);
+  const evmTokens = useAllEvmAccountTokenSlugs(accountEvmAddress);
+
+  return useMemo(() => [...tezTokens, ...evmTokens], [tezTokens, evmTokens]);
+};
+
+export const useEnabledTezosAccountTokenSlugs = (publicKeyHash: string) => {
+  const tokens = useTezosAccountTokens(publicKeyHash);
+
+  return useMemo(
+    () =>
+      tokens.reduce<string[]>((acc, { slug, status, chainId }) => {
+        if (status === 'enabled') acc.push(toChainAssetSlug(TempleChainKind.Tezos, chainId, slug));
+
+        return acc;
+      }, []),
+    [tokens]
+  );
+};
+
+export const useAllTezosAccountTokenSlugs = (publicKeyHash: string) => {
+  const tokens = useTezosAccountTokens(publicKeyHash);
+
+  return useMemo(
+    () =>
+      tokens.reduce<string[]>((acc, { slug, status, chainId }) => {
```

Comment:
`tokens.filter(...).map(...)` must do the same job. However, if it is slower, please create a `filterMap` function with such implementation and use it at all places like this in this file and in `src/lib/assets/hooks/collectibles.ts`